sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,851 @@
+    var options: EventRoutesAddOptionalParams = {};
+    var eventRoute: EventRoute = {
+      endpointName,
+      filter
+    };
+    options.eventRoute = eventRoute;
+    return this.client.eventRoutes.add(eventRouteId, options);
+  }
+
+  /**
+   * Delete an event route.
+   *
+   * @param eventRouteId The Id of the model to delete.
+   * @returns The http response.
+   */
+  public deleteEventRoute(eventRouteId: string): Promise<RestResponse> {
+    return this.client.eventRoutes.deleteMethod(eventRouteId);
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the pagination of {@link query}.
+   *
+   * @param {PageSettings} continuationState An object that indicates the position of the paginated request.
+   * @param {RequestOptionsBase} [options] Common options for the iterative endpoints.
+   *
+   */
+  private async *queryTwinsPage(
+    options: RequestOptionsBase,
+    continuationState: PageSettings
+  ): AsyncIterableIterator<QueryQueryTwinsResponse> {
+    if (continuationState.continuationToken == null) {
+      const optionsComplete: QuerySpecification = {
+        continuationToken: continuationState.continuationToken,
+        ...options
+      };
+      const queryResponse = await this.client.query.queryTwins(optionsComplete);
+      continuationState.continuationToken = queryResponse.continuationToken;
+      yield queryResponse;
+    }
+    while (continuationState.continuationToken) {
+      const optionsNext: QuerySpecification = {
+        continuationToken: continuationState.continuationToken,
+        ...options
+      };
+      const queryResponse = await this.client.query.queryTwins(optionsNext, options);
+
+      continuationState.continuationToken = queryResponse.continuationToken;
+      yield queryResponse;
+    }
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the iteration of all the available results of {@link query}.
+   * @param {RequestOptionsBase} [options] Common options for the iterative endpoints.
+   */
+  private async *queryTwinsAll(options: RequestOptionsBase): AsyncIterableIterator<any> {
+    const f = {};
+
+    for await (const page of this.queryTwinsPage(options, f)) {
+      if (page.items) {
+        for (const item of page.items) {
+          yield item;
+        }
+      }
+    }
+  }
+
+  /**
+   * Query for digital twins.
+   *
+   * @param query The query string, in SQL-like syntax.
+   * @returns The pageable list of query results.
+   */
+  public queryTwins(query?: string): PagedAsyncIterableIterator<any, QueryQueryTwinsResponse> {
+    var querySpecification = <QuerySpecification>{};
+    querySpecification.query = query;
+
+    const iter = this.queryTwinsAll(querySpecification);
+
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) => this.queryTwinsPage(querySpecification, settings)
+    };
+  }
+}
+
+export {
```

Comment:
typically we have a `/src/index.ts` that handles all top level package exports instead of the client doing this

Reply:
It is in the ToDo list

---
In reply to: [447263479](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r447263479) [](ancestors = 447263479)